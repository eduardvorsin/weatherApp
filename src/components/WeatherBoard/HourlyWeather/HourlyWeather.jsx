import PropTypes from 'prop-types';
import { getIconPathByCode, getWeatherStatusByCode } from '../../../utils/utils';
import classes from './HourlyWeather.module.css';

export default function HourlyWeather({
  temperature,
  weatherCode,
  className,
  testId,
}) {
  const iconPath = getIconPathByCode(weatherCode);
  const status = getWeatherStatusByCode(weatherCode);

  return (
    <div
      className={`${classes['hourly-weather']} ${className}`}
      data-testid={testId}
    >
      <img
        className={classes['hourly-weather__icon']}
        src={iconPath}
        alt={status}
      />
      <p
        className={classes['hourly-weather__temperature']}
      >
        {temperature}
        Â°
      </p>
      <p
        className={classes['hourly-weather__status']}
      >
        {status}
      </p>
    </div>
  );
}

HourlyWeather.defaultProps = {
  temperature: 0,
  weatherCode: 0,
  className: '',
  testId: '',
};

HourlyWeather.propTypes = {
  className: PropTypes.string,
  temperature: PropTypes.number,
  weatherCode: PropTypes.number,
  testId: PropTypes.string,
};
