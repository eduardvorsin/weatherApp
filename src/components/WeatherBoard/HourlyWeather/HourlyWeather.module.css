.hourly-weather>*:not(:last-child) {
  margin-bottom: 10px;
}

.hourly-weather__icon {
  width: 80px;
  height: 80px;
}


.hourly-weather__status {
  font-weight: 700;
}

.hourly-weather__temperature,
.hourly-weather__status {
  margin: 0;
}
