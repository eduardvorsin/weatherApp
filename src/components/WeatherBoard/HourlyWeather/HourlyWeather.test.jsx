import { render, screen } from '@testing-library/react';
import HourlyWeather from './HourlyWeather';

describe('HourlyWeather component tests', () => {
  it('renders correctly', () => {
    render(<HourlyWeather
      className="test-class"
      weatherCode={0}
      temperature={-30}
    />);

    expect(screen.getByText(/-30°/)).toBeInTheDocument();
    expect(screen.getByText(/ясно/)).toBeInTheDocument();
  });

  it('renders without props', () => {
    render(<HourlyWeather />);

    expect(screen.getByText(/ясно/)).toBeInTheDocument();
    expect(screen.getByText(/0°/)).toBeInTheDocument();
  });

  it('the classname is correctly assigned to the element', () => {
    render(<HourlyWeather
      className="test-class"
      testId="hourly-weather"
    />);

    expect(screen.getByTestId('hourly-weather')).toHaveClass('test-class');
  });

  it('snapshot with default data', () => {
    render(<HourlyWeather
      testId="hourly-weather"
    />);

    expect(screen.getByTestId('hourly-weather')).toMatchSnapshot();
  });

  it('snapshot with mock text', () => {
    render(<HourlyWeather
      className="test-class"
      weatherCode={0}
      temperature={-30}
      testId="hourly-weather"
    />);

    expect(screen.getByTestId('hourly-weather')).toMatchSnapshot();
  });
});
