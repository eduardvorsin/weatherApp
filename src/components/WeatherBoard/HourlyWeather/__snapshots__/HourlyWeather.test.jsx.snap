// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HourlyWeather component tests snapshot with default data 1`] = `
<div
  class="hourly-weather "
  data-testid="hourly-weather"
>
  <img
    alt="ясно"
    class="hourly-weather__icon"
    src="/assets/icons/sunny.svg"
  />
  <p
    class="hourly-weather__temperature"
  >
    0
    °
  </p>
  <p
    class="hourly-weather__status"
  >
    ясно
  </p>
</div>
`;

exports[`HourlyWeather component tests snapshot with mock text 1`] = `
<div
  class="hourly-weather test-class"
  data-testid="hourly-weather"
>
  <img
    alt="ясно"
    class="hourly-weather__icon"
    src="/assets/icons/sunny.svg"
  />
  <p
    class="hourly-weather__temperature"
  >
    -30
    °
  </p>
  <p
    class="hourly-weather__status"
  >
    ясно
  </p>
</div>
`;
